src/stcal/ramp_fitting/likely_fit.py

```diff
@@ -0,0 +1,1274 @@
+        # No need for indexing on the first pass.
+        if j == 0:
+            result = fit_ramps(
+                loc_diff,
+                covar,
+                gain,
+                rnoise,
+                count_rate_guess=count_rate_guess,
+                diffs2use=diffs2use,
+                detect_jumps=True,
+            )
+            # Also save the count rates so that we can use them later
+            # for debiasing.
+            countrate = result.countrate * 1.0
+        else:
+            result = fit_ramps(
+                loc_diff[:, recheck],
+                covar,
+                gain[recheck],
+                rnoise[recheck],
+                count_rate_guess=count_rate_guess[:, recheck],
+                diffs2use=diffs2use[:, recheck],
+                detect_jumps=True,
+            )
+
+        # Chi squared improvements
+        dchisq_two = result.chisq - result.chisq_two_omit
+        dchisq_one = result.chisq - result.chisq_one_omit
+
+        # We want the largest chi squared difference
+        best_dchisq_one = np.amax(dchisq_one * one_omit_ok[:, np.newaxis], axis=0)
+        best_dchisq_two = np.amax(
+            dchisq_two * two_omit_ok[:, np.newaxis], axis=0
+        )
+
+        # Is the best improvement from dropping one resultant
+        # difference or two?  Two drops will always offer more
+        # improvement than one so penalize them by the respective
+        # thresholds.  Then find the chi squared improvement
+        # corresponding to dropping either one or two reads, whichever
+        # is better, if either exceeded the threshold.
+        onedropbetter = (
+            best_dchisq_one - threshold_one_omit > best_dchisq_two - threshold_two_omit
+        )
+
+        best_dchisq = (
+            best_dchisq_one * (best_dchisq_one > threshold_one_omit) * onedropbetter
+        )
+        best_dchisq += (
+            best_dchisq_two * (best_dchisq_two > threshold_two_omit) * (~onedropbetter)
+        )
+
+        # If nothing exceeded the threshold set the improvement to
+        # NaN so that dchisq==best_dchisq is guaranteed to be False.
+        best_dchisq[best_dchisq == 0] = np.nan
+
+        # Now make the masks for which resultant difference(s) to
+        # drop, count the number of ramps affected, and drop them.
+        # If no ramps were affected break the loop.
+        dropone = dchisq_one == best_dchisq
+        droptwo = dchisq_two == best_dchisq
+
+        drop = np.any([np.sum(dropone, axis=0), np.sum(droptwo, axis=0)], axis=0)
+
+        if np.sum(drop) == 0:
+            break
+
+        # Store the updated counts with omitted reads
+        new_cts = np.zeros(np.sum(recheck))
+        i_d1 = np.sum(dropone, axis=0) > 0
+        new_cts[i_d1] = np.sum(result.countrate_one_omit * dropone, axis=0)[i_d1]
+        i_d2 = np.sum(droptwo, axis=0) > 0
+        new_cts[i_d2] = np.sum(result.countrate_two_omit * droptwo, axis=0)[i_d2]
```

Comment:
Testing locally, this line often fills the log with "invalid value" warning messages.  Can we please catch and hide the warning?